pkg/liqo-controller-manager/virtualnode-controller/virtualkubelet.go

```diff
@@ -177,15 +186,36 @@ func (r *VirtualNodeReconciler) getVirtualKubeletDeployment(
 	return &deployList.Items[0], nil
 }
 
-func checkVirtualKubeletPodAbsence(ctx context.Context, cl client.Client,
-	vn *virtualkubeletv1alpha1.VirtualNode, vkopt *vkforge.VirtualKubeletOpts) (bool, error) {
+func (r *VirtualNodeReconciler) checkVirtualKubeletPodAbsence(ctx context.Context,
+	vn *virtualkubeletv1alpha1.VirtualNode) (bool, error) {
 	klog.Warningf("[%v] checking virtual-kubelet pod absence", vn.Spec.ClusterIdentity.ClusterID)
-	list := &corev1.PodList{}
-	labels := vkforge.VirtualKubeletLabels(vn, vkopt)
-	err := cl.List(ctx, list, client.MatchingLabels(labels))
+	list, err := getters.ListVirtualKubeletPodsFromVirtualNode(ctx, r.Client, vn, r.VirtualKubeletOptions)
 	if err != nil {
-		klog.Error(err)
-		return false, nil
+		return false, err
 	}
 	return len(list.Items) == 0, err
 }
+
+func (r *VirtualNodeReconciler) checkVirtualKubeletPodsContainsOnlyArg(
```

Comment:
Can you choose a better function naming and/or add some comments on what it is doing?